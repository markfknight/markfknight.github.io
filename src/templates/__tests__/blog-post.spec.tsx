import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';

import BlogPost, { BlogPostProps } from '../blog-post';

jest.mock(`../../components/seo`);

describe(`Blog Post snapshot`, () => {
  it(`renders correctly`, () => {
    const props: PropsWithChildren<BlogPostProps> = {
      location: {
        pathname: ``,
      },
      data: {
        site: {
          siteMetadata: {
            title: ``,
          },
        },
        markdownRemark: {
          id: ``,
          excerpt: ``,
          html: ``,
          fields: {
            slug: ``,
          },
          frontmatter: {
            title: ``,
            date: ``,
          },
        },
      },
      pageContext: {
        previous: {
          id: ``,
          excerpt: ``,
          html: ``,
          fields: {
            slug: ``,
          },
          frontmatter: {
            title: ``,
            date: ``,
          },
        },
        next: {
          id: ``,
          excerpt: ``,
          html: ``,
          fields: {
            slug: ``,
          },
          frontmatter: {
            title: ``,
            date: ``,
          },
        },
      },
    } as PropsWithChildren<BlogPostProps>;
    const tree = render(<BlogPost {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
