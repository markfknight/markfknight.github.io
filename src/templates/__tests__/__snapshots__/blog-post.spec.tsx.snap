// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Post snapshot renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container max-w-3xl px4 py10"
      >
        <header
          class="py-10"
        >
          <h3>
            <a
              href="/"
            />
          </h3>
        </header>
        <main>
          <article
            class="prose prose-sm sm:prose lg:prose-lg xl:prose-2xl mx-auto"
          >
            <header>
              <h1 />
              <p
                class="text-gray-500"
              />
            </header>
            <section
              class="text-justify"
            />
            <footer />
          </article>
          <nav>
            <ul
              style="display: flex; flex-wrap: wrap; justify-content: space-between; list-style: none; padding: 0px;"
            >
              <li>
                <a
                  href=""
                  rel="prev"
                  style="box-shadow: none; text-decoration: none;"
                >
                  ← 
                  
                </a>
              </li>
              <li>
                <a
                  href=""
                  rel="next"
                  style="box-shadow: none; text-decoration: none;"
                >
                  
                   →
                </a>
              </li>
            </ul>
          </nav>
        </main>
        <footer
          class="py-10 text-center"
        >
          © 
          2021
          , Built with
           
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container max-w-3xl px4 py10"
    >
      <header
        class="py-10"
      >
        <h3>
          <a
            href="/"
          />
        </h3>
      </header>
      <main>
        <article
          class="prose prose-sm sm:prose lg:prose-lg xl:prose-2xl mx-auto"
        >
          <header>
            <h1 />
            <p
              class="text-gray-500"
            />
          </header>
          <section
            class="text-justify"
          />
          <footer />
        </article>
        <nav>
          <ul
            style="display: flex; flex-wrap: wrap; justify-content: space-between; list-style: none; padding: 0px;"
          >
            <li>
              <a
                href=""
                rel="prev"
                style="box-shadow: none; text-decoration: none;"
              >
                ← 
                
              </a>
            </li>
            <li>
              <a
                href=""
                rel="next"
                style="box-shadow: none; text-decoration: none;"
              >
                
                 →
              </a>
            </li>
          </ul>
        </nav>
      </main>
      <footer
        class="py-10 text-center"
      >
        © 
        2021
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
